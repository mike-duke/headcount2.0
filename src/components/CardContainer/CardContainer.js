import React from 'react';
import './CardContainer.scss';
import Card from '../Card/Card.js';
import CompareContainer from '../CompareContainer/CompareContainer.js';
import PropTypes from 'prop-types';

const CardContainer = ({districtData, compareMessage, districtsToCompare}) => {
  const cards = districtData.map((district) => {
    return <Card location={district.location}
                  stats={district.stats}
                  key={district.location} />;
  });
  
  return (
    <section className="card-section">
      <h1 className="title-header">Kindergartners In Full Day Programs</h1>
        <CompareContainer compareMessage={compareMessage}
                          districtsToCompare={districtsToCompare} />

        <article className="card-container">
        {cards}
      </article>
    </section>
  );
}

CardContainer.propTypes = {
  districtData: PropTypes.array.isRequired,
  compareMessage: PropTypes.string.isRequired,
  districtsToCompare: PropTypes.array.isRequired
}

export default CardContainer;