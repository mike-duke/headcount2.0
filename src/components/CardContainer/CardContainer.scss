section {
  text-align: center;
}

.title-header {
  z-index: 1;
}

.card-container {
  display: flex;
  z-index: 1;
  flex-wrap: wrap;
  justify-content: space-around;
  position: relative;
}