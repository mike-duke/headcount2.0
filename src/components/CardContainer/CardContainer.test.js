import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer.js';

describe('CardContainer', () => {

  it('should match the snapshot with no cards passed down', () => {
    let wrapper = shallow(<CardContainer districtData={[]} />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot with cards passed down', () => {
    let wrapper = shallow(<CardContainer districtData={[]} />);

    expect(wrapper).toMatchSnapshot();
  })
})