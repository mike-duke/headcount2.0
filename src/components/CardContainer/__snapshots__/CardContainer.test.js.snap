// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with cards passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districtData={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title-header"
        >
          Kindergartners In Full Day Programs
        </h1>,
        <article
          className="card-container"
        />,
      ],
      "className": "card-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Kindergartners In Full Day Programs",
          "className": "title-header",
        },
        "ref": null,
        "rendered": "Kindergartners In Full Day Programs",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title-header"
          >
            Kindergartners In Full Day Programs
          </h1>,
          <article
            className="card-container"
          />,
        ],
        "className": "card-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Kindergartners In Full Day Programs",
            "className": "title-header",
          },
          "ref": null,
          "rendered": "Kindergartners In Full Day Programs",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot with no cards passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districtData={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title-header"
        >
          Kindergartners In Full Day Programs
        </h1>,
        <article
          className="card-container"
        />,
      ],
      "className": "card-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Kindergartners In Full Day Programs",
          "className": "title-header",
        },
        "ref": null,
        "rendered": "Kindergartners In Full Day Programs",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title-header"
          >
            Kindergartners In Full Day Programs
          </h1>,
          <article
            className="card-container"
          />,
        ],
        "className": "card-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Kindergartners In Full Day Programs",
            "className": "title-header",
          },
          "ref": null,
          "rendered": "Kindergartners In Full Day Programs",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
